### ISSUE TITLE
Build args passed in wrong order to devspaces-build init container

### ISSUE THREAD

**Issue by markphillips100** [Azure/dev-spaces#93]
Since having used `azds up --clean` the docker file no longer builds.

I've run `kubectl logs <mypod> -c devspaces-build` to check the reason for the failure and the logged "build args:" value contains a long line of comma-separated arguments of which this is a snippet: `/Dockerfile.develop,.,--no-cache,--pull`.  Note: the PATH . does not appear last which I think is a requirement for using `docker build`.

In addition, I've executed `azds down` which clears the pod, however, even running just `azds up` (without the clean this time) still fails to build and the same `--no-cache,--pull` appears in the log.  It seems the build service is caching the --clean or at least appears that way.

```
azds --version
Azure Dev Spaces CLI
1.0.20190423.8
API v3.2

AKS 1.12.7
```

**Comment by xinyanmsft** [comment]
thanks for reporting, Mark. Unfortunately I couldn't reproduce your problem. 'azds up --clean' indeed put the --no-cache and --pull options at the end, but 'docker build' appears to be working Okay for me. 

however in your snippet, '/Dockerfile.develop' does concern me since the path of Dockerfile should not be in the root directory. 

to help us troubleshoot, can you 
1. send us your azds.yaml file?
2. post the whole log from 'kubectl logs <mypod> -c devspaces-build'?

thanks!

**Comment by markphillips100** [comment]
My fault @xinyanmsft.  I'm setting a build arg (AZDS_NUGET_FEED) in the azds.yaml where the environment value is a json string.  Like so:
```
      args:
        BUILD_CONFIGURATION: ${BUILD_CONFIGURATION:-Debug}
        AZDS_NUGET_FEED: ${AZDS_NUGET_FEED:-}
```
The json value caused parsing issues I guess so I wrapped in escaped doublequotes and all is now good:
```
      args:
        BUILD_CONFIGURATION: ${BUILD_CONFIGURATION:-Debug}
        AZDS_NUGET_FEED: \"${AZDS_NUGET_FEED:-}\"
```
By the way, the Dockerfile.develop path is `-f,src/MyProject/Dockerfile.develop`.  I just cut the snippet short.

**Comment by ShrutiPPathak** [comment]
@markphillips100 : Based on your feedback, we identified an issue in our code where we were quoting arguments unnecessarily. I just submitted a fix for it and I expect it to start rolling out in a couple of weeks. After the fix, you will have to update your azds.yaml as you no longer need to wrap json values in escaped doublequotes. Apologies for the inconvenience and thanks again for your valuable feedback.
